quickwit-aws/src/retry.rs

```diff
@@ -71,10 +71,23 @@ impl<E> IsRetryable for Retry<E> {
     }
 }
 
-// TODO define retry strategy
-/// Retry with exponential backoff and full jitter. Implementation and default values originate from
-/// the Java SDK. See also: <https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/>.
-pub async fn retry<F, U, E, Fut>(f: F) -> Result<U, E>
+pub struct RetryParams {
+    base_delay: Duration,
+    max_delay: Duration,
+    max_attempts: usize,
+}
+
+impl Default for RetryParams {
+    fn default() -> Self {
+        Self {
+            base_delay: BASE_DELAY,
+            max_delay: MAX_DELAY,
+            max_attempts: MAX_RETRY_ATTEMPTS,
```

Comment:
Let's set the `DEFAULT_MAX_RETRY_ATTEMPTS` to 30 and use the value `3` for the S3 storage impl.